journal.py

```diff
@@ -102,69 +155,90 @@ def update_entry(request, id):
     request.db.cursor().execute(UPDATE_ENTRY, [title, text, id])
 
 
-def write_entry(request):
-    """ add a new entry to the database"""
-    title = request.params.get('title', None)
-    text = request.params.get('text', None)
-    created = datetime.datetime.utcnow()
-    request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
+# def write_entry(request):
+#     """ add a new entry to the database"""
+#     title = request.params.get('title', None)
+#     text = request.params.get('text', None)
+#     created = datetime.datetime.utcnow()
+#     request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
 
 
-@view_config(route_name='add', request_method='POST')
+@view_config(route_name='add', renderer='json')
 def add_entry(request):
-    try:
-        write_entry(request)
-    except psycopg2.Error:
-        # this will catch any errors generated by the database
-        return HTTPInternalServerError
-    return HTTPFound(request.route_url('home'))
-
+    if request.authenticated_userid:
+        if request.method == 'POST':
+            try:
+                Entry.from_request(request)
+            # This error type needs to be updated since it will now be an SQLA error
+            # type.
+            except psycopg2.Error:
```

Comment:
You won't get psycopg2 errors anymore, now that you are using SQLAlchemy.  This should be updated to reflect the new possible problems.